app/src/main/java/com/example/android_level_3/FragmentAddContact.kt

```diff
@@ -0,0 +1,219 @@
+            if (visible) {
+                with(binding){
+                    toolbarAddContact.containerTextAddContact.visibility = View.GONE
+                    toolbarAddContact.containerSearchAddContact.visibility = View.VISIBLE
+                }
+            } else {
+                with(binding){
+                    toolbarAddContact.containerTextAddContact.visibility = View.VISIBLE
+                    toolbarAddContact.containerSearchAddContact.visibility = View.GONE
+                    toolbarAddContact.edSearchAddContact.setText("")
+                }
+            }
+        }
+
+        // "слушатель" на добавление контакта
+        val resultOfAddingContact =
+            findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<Int>(Const.RESULT_KEY)
+
+        // наблюдатель за изменением состояния "слушателя"
+        resultOfAddingContact?.observe(viewLifecycleOwner) { id ->
+            if (id != null) {
+                contactId = id
+                requestAddToContactList()
+            }
+        }
+    }
+
+    private fun requestAddToContactList() {
+        viewModel.addToContactList(
+            userId = viewModel.userId.value!!, token = viewModel.token.value!!,
+            contactId = contactId!!, progressBar = isVisibleProgressBar)
+    }
+
+    private fun requestGetAllUsers() {
+        viewModel.getAllUsers(token = viewModel.token.value!!, progressBar = isVisibleProgressBar)
+    }
+
+    private fun setListeners() {
+
+        binding.toolbarAddContact.imgBackAddContact.setOnClickListener {
+            findNavController().popBackStack()
+        }
+
+        binding.toolbarAddContact.imgSearchAddContact.setOnClickListener {
+            viewModel.isActiveSearchAddContact.value = true
+        }
+
+        binding.toolbarAddContact.imgCloseAddContact.setOnClickListener {
+            viewModel.isActiveSearchAddContact.value = false
+        }
+
+        binding.toolbarAddContact.edSearchAddContact.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) { }
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) { }
+            override fun afterTextChanged(s: Editable?) {
+                if(s?.isNotEmpty() == true ) {
+                    viewModel.filteredUserList.value?.clear()
+
+                    val filteredList = viewModel.listOfAllUsers.value?.filter {
+                        it.name?.contains(s, true) == true
+                    }
+                    viewModel.filteredUserList.value = filteredList?.toMutableList()
+                    afterFilterAction()
+                } else {
+                    if (viewModel.listOfAllUsers.value?.isNotEmpty() == true)
+                        viewModel.listOfAllUsers.value = viewModel.listOfAllUsers.value
+                }
+            }
+        })
+    }
+
+    private fun afterFilterAction() {
```

Comment:
function name should start with a verb!
I suggest you to install Ktlinter in the project